File: packages/shared/src/types/facture.ts
import { BaseEntity, UUID } from './common';
import { LigneDevis } from './devis';

export type FactureStatut = 'brouillon' | 'envoyee' | 'payee' | 'annulee';

export interface Facture extends BaseEntity {
  devisId: UUID;
  numero: string; // ex: FAC-2025-001
  // Dates
  dateEmission: Date;
  dateEcheance: Date;
  datePaiement?: Date;
  // Contenu (copie du devis au moment de la facturation)
  lignes: LigneDevis[];
  // Totaux
  totalHT: number;
  totalTVA: number;
  totalTTC: number;
  // Paiement
  statut: FactureStatut;
  modePaiement?: 'virement' | 'cheque' | 'especes' | 'carte';
  referencePaiement?: string;
  // PDF
  pdfUrl?: string;
  // Notes
  mentionsLegales?: string;
  notes?: string;
}

export interface CreateFactureDto {
  devisId: UUID;
  dateEcheance: Date;
  mentionsLegales?: string;
  notes?: string;
}

export interface UpdateFactureDto {
  statut?: FactureStatut;
  datePaiement?: Date;
  modePaiement?: 'virement' | 'cheque' | 'especes' | 'carte';
  referencePaiement?: string;
  notes?: string;
}

File: apps/mobile/lib/domain/models/facture.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../enums/facture_statut.dart';
import '../enums/mode_paiement.dart';

part 'facture.freezed.dart';
part 'facture.g.dart';

@freezed
abstract class Facture with _$Facture {
  const factory Facture({
    required String id,
    required String devisId,
    required String numero,
    required DateTime dateEmission,
    required DateTime dateEcheance,
    DateTime? datePaiement,
    required double totalHT,
    required double totalTVA,
    required double totalTTC,
    @Default(FactureStatut.brouillon) FactureStatut statut,
    ModePaiement? modePaiement,
    String? referencePaiement,
    String? pdfUrl,
    String? mentionsLegales,
    String? notes,
    DateTime? deletedAt,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _Facture;

  factory Facture.fromJson(Map<String, dynamic> json) =>
      _$FactureFromJson(json);
}
